tensorboard/backend/application.py

```diff
@@ -371,18 +305,23 @@ def _reload_forever():
   return thread
 
 
-def get_tensorboard_tag():
-  """Read the TensorBoard TAG number, and return it or an empty string."""
-  return hacky_loader('TAG').strip()
-
+def get_default_assets_zip_provider():
+  """Opens stock TensorBoard web assets collection.
 
-def hacky_loader(path):
-  """Load an asset from the tensorboard directory, in a platform-generic way."""
-  try:
-    return tf.resource_loader.load_resource(path)
-  except IOError:
-    tensorboard_root = os.path.join(os.path.dirname(__file__), os.pardir)
-    path = os.path.join(tensorboard_root, path)
-    path = os.path.abspath(path)
-    with open(path, 'rb') as f:
-      return f.read()
+  Returns:
+    Returns function that returns a zipfile.ZipFile instance containing files
+    with absolute web paths, or None if webfiles.zip could not be found.
+  """
+  path = os.path.join(
+      tf.resource_loader.get_data_files_path(), os.pardir, 'webfiles.zip')
+  if not os.path.exists(path):
+    tf.logging.warning('webfiles.zip static assets not found: %s', path)
+    return None
+  return lambda: zipfile.ZipFile(open(path))
```

Comment:
It looks like this might be a resource/fd leak? I'm not sure (rarely used `zipfile`); please correct me if I'm wrong.
```py
>>> import zipfile
>>> infile = open('greetings.zip')
>>> zf = zipfile.ZipFile(infile)
>>> zf.close()
>>> infile.closed
False
>>> infile.close()  # but clients of get_default_assets_zip_provider can't do this
>>> infile.closed
True
```